docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+> *not* the new shimv2 runtime instance (`containerd-shim-kata-v2` binary).
+
+The diagram below shows how the original architecture was simplified
+with the advent of shimv2.
 
 ![Kubernetes integration with shimv2](arch-images/shimv2.svg)
 
-The container process is then spawned by
-[`kata-agent`](../../src/agent), an agent process running
-as a daemon inside the virtual machine. `kata-agent` runs a [`ttRPC`](https://github.com/containerd/ttrpc-rust) server in
-the guest using a VIRTIO serial or VSOCK interface which QEMU exposes as a socket
-file on the host. `shimv2` uses a `ttRPC` protocol to communicate with
-the agent. This protocol allows the runtime to send container management
-commands to the agent. The protocol is also used to carry the I/O streams (stdout,
-stderr, stdin) between the containers and the manage engines (e.g. CRI-O or containerd).
-
-For any given container, both the init process and all potentially executed
-commands within that container, together with their related I/O streams, need
-to go through the VSOCK interface exported by QEMU.
-
-The container workload, that is, the actual OCI bundle rootfs, is exported from the
-host to the virtual machine.  In the case where a block-based graph driver is
-configured, `virtio-scsi` will be used. In all other cases a `virtio-fs` VIRTIO mount point
-will be used. `kata-agent` uses this mount point as the root filesystem for the
-container processes.
+## System overview
 
-## Virtualization
+### Container creation
+
+The steps below show at a high level how a Kata Container container is
+created using the containerd container manager:
+
+1. The user requests the creation of a container by running `ctr run`.
+1. The container manager daemon runs a single instance of the Kata [runtime](#runtime).
+1. The Kata runtime loads its [configuration file](#configuration).
+1. The container manager calls a set of shimv2 API functions on the runtime.
+1. The Kata runtime launches the configured [hypervisor](#hypervisor).
+1. The hypervisor creates a VM using the [guest assets](#guest-assets).
+1. The hypervisor boots (starts) the VM.
+1. The [agent](#agent) is launched inside the VM.
+1. The agent creates a container environment to host the [workload](#workload).
+1. The agent spawns the workload inside the container environment.
+1. Control is returned to the user running `ctr`.
+
+   If the user requested it, the standard I/O streams of the workload
+   running inside the container in the VM are connected to the user's
+   terminal.
+
+Further details of these steps are provided in the sections below.
+
+### Container shutdown
 
-How Kata Containers maps container concepts to virtual machine technologies, and how this is realized in the multiple
-hypervisors and VMMs that Kata supports is described within the [virtualization documentation](./virtualization.md)
+There are two possible ways for the container environment to be
+destroyed:
+
+- When the [workload](#workload) exits.
```

Comment:
It might be worth calling out which one is a graceful exit and which one isn't. It is kind of implied but when you state that there are two ways for it to be destroyed I can think of other ways like the computer rebooting.   Also, destroyed sounds kind of harsh. Perhaps say terminated instead.